scylla/src/routing.rs

```diff
@@ -35,6 +35,36 @@ pub struct ShardInfo {
     msb_ignore: u8,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Keyspace {
+    // replication_factor might be None ex. with NetworkTopologyStrategy
+    pub replication_factor: Option<usize>,
+    // TODO - is strategy_class required for each keyspace to exist? Maybe Option is not needed
+    pub strategy_class: Option<Strategy>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Strategy {
+    SimpleStrategy,
+    LocalStrategy,
+    NetworkTopologyStrategy,
+    // TODO - add more strategies
+    WithName(String),
+}
+
+impl Strategy {
+    pub fn from_string(strategy_string: String) -> Strategy {
```

Comment:
Any reason why you don't implement the FromStr trait for this function? This function takes a String and than converts it back to a &str, which makes the String parameter an unnecessary allocation.

Reply:
I wanted to avoid having to clone `strategy_string` when creating `Strategy::WithName`. There is no need to turn `&str` to `String` before calling this function because we do `HashMap<String, String>::remove()` which moves string with strategy name out of the map

Reply:
If that will be the only place from which it's called, it's a good reason to have the `String` parameter indeed